### DISCUSSION TITLE
[Error] Bad Url when the server is behind a revere proxy with different port

### DISCUSSION THREAD

**Discussion by a4amado** [nextauthjs/next-auth#12412] [Help]
I have two servers
1. express server port 3001
2. nextjs server port 3000
3. 
and nginx reverse proxy server 5000

when logging, nextjs throws error `Bad URL` bc the internal url `http://localhost:3000` is dofferent from the exposed port `http://localhost:5000`

I tries to over write  it using `
NEXT_AUTH_URL="http://localhost:5000"
NEXTAUTH_URL_INTERNAL="http://localhost:5000"
`
but with no luck

**Comment by MasterProvider** [comment]
Same as my experience #12887
